spigot/src/main/java/com/github/camotoy/geyserblockjavaplayers/spigot/GeyserBlockJavaPlayers.java

```diff
@@ -36,7 +36,8 @@ public void onDisable() {
     @EventHandler
     public void onPlayerJoin(PlayerJoinEvent event) {
         boolean isBedrockPlayer = this.playerChecker.isBedrockPlayer(event.getPlayer().getUniqueId());
-        if (!isBedrockPlayer) {
+        boolean hasPermissionBypass = event.getPlayer().hasPermission("javablock.bypass");
```

Comment:
The way I would implement this is: check the permission first, and if the player has permission, don't even bother trying to determine if the player if Bedrock, just return there. https://github.com/Camotoy/GeyserPreventServerSwitch/blob/master/src/main/java/com/github/camotoy/geyserpreventserverswitch/Events.java#L19

Also, to prevent possible conflicts, I would recommend the permission string has the full plugin name: `geyserblockjavaplayers.bypass`.